.github/workflows/ci.yml

```diff
@@ -49,6 +49,26 @@ jobs:
           override: true
       - run: cargo test --workspace -- --nocapture
 
+  examples:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly-2022-01-01
+    steps:
+      - uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5 # v2
+        with:
+          submodules: true
+      - uses: actions-rs/toolchain@63eb9591781c46a70274cb3ebdf190fce92702e8 # v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          components: rustfmt
+          override: true
+      - run: for eg in `ls examples/* | xargs basename --suffix=.rs`; do cargo run --example ${eg}; done
```

Comment:
why not just `examples/*.rs`?

Reply:
GitHub shows me: 
```
  - run: for eg in `ls examples/*.rs | xargs basename --suffix=.rs`; do cargo run --example ${eg}; done
```
Is that not what you see?  Or did I misunderstand what you wanted?

Reply:
I was being dumb. The file is `examples/signature.rs` but you need to run `cargo run --example signature`. So I think what you have now is the best possible.

I think Cargo got this wrong; --example should have taken a filename, not an identifier. This would also allow examples to be put into subfolders.

Reply:
Yeah, or something like https://github.com/rust-lang/cargo/issues/8356 would help.

Reply:
This will run all examples: `cargo test --examples`.  As will `cargo test --all`

Reply:
Thanks!  I'm not sure how I failed to find that option when I went looking &ndash; I guess I found that https://github.com/rust-lang/cargo/issues/8356 was open so assumed it wasn't possible.